test/torchaudio_unittest/transforms/autograd_test_impl.py

```diff
@@ -76,6 +76,26 @@ def test_spectrogram(self, kwargs):
         waveform = get_whitenoise(sample_rate=8000, duration=0.05, n_channels=2)
         self.assert_grad(transform, [waveform], nondet_tol=1e-10)
 
+    @parameterized.expand([
+        ({'pad': 0, 'normalized': False, 'power': None, 'return_complex': True}, ),
+        ({'pad': 3, 'normalized': False, 'power': None, 'return_complex': True}, ),
+        ({'pad': 0, 'normalized': True, 'power': None, 'return_complex': True}, ),
+        ({'pad': 3, 'normalized': True, 'power': None, 'return_complex': True}, ),
+        ({'pad': 0, 'normalized': False, 'power': None}, ),
+        ({'pad': 3, 'normalized': False, 'power': None}, ),
+        ({'pad': 0, 'normalized': True, 'power': None}, ),
+        ({'pad': 3, 'normalized': True, 'power': None}, ),
+    ])
+    def test_inverse_spectrogram(self, kwargs):
+        # create a realistic input:
+        transform = T.Spectrogram(**kwargs)
```

Comment:
So, there's the length option for itsft that's supposed to help with the problem you discuss. So even though your script fails, 
`import torch
import torchaudio

theta = torch.linspace(0, 300 * 3.14, 8000)
waveform = torch.sin(theta).unsqueeze(0)
torchaudio.save('original.wav', waveform, 8000)
print(waveform.shape)

n_fft = 400
for center in [True, False]:
    spec = torch.stft(waveform, n_fft=n_fft, center=center, return_complex=True)
    print('spec:', spec.shape, spec.dtype)
    recon = torch.istft(spec, n_fft=n_fft, center=center, length=8000). # note length here
    torchaudio.save(f'recon_center_{center}.wav', recon, 8000)
    print('recon:', recon.shape)
    torch.testing.assert_allclose(waveform, recon)
`works just fine.
However, this is not the case with T.spectrogram/inverse_spectrogram.

Reply:
So here's what actually breaks. Take your example and do this:
```
    window = torch.hann_window(n_fft)
```
and pass 'window=window' to stft and istft. That breaks.

Reply:
It's nice that `length` parameter can resolve this, however, from user's of `torchaudio`, this may be inconvenient.
The following is the scenario I have in my mind.

Imagine that we have some NN model that performs a task in frequency domain (enhancement, noise reduction, separation etc...)

```
waveform -> spectrogram -> NN model -> spectrogram -> waveform
```

If our implementation needs `length` to be passed around, so as to recover the waveform with the same length, we need to pass `length` parameter separately, in addition to the spectrogram tensor.

```
waveform -> spectrogram -> NN model -> spectrogram -> waveform
    |                                                 ↑
    └--------------------> length --------------------┘ 
```

Reply:
For `window`, I was hoping/expecting that `istft` can do it but, it looks like that's not the case. At the moment, I do not have an idea for workaround. I would say we can comeback later for this.

Reply:
> So here's what actually breaks. Take your example and do this:
> 
> ```
>     window = torch.hann_window(n_fft)
> ```
> 
> and pass 'window=window' to stft and istft. That breaks.

The reason is that the hann window fails the non-zero check in istft:
```
# check NOLA non-zero overlap condition
    window_envelop_lowest = window_envelop.abs().min()
    assert window_envelop_lowest > 1e-11, "window overlap add min: %f" % (
        window_envelop_lowest
    )
```
In librosa, instead of checking no_zero condition in window envelop, they apply normalization to [non-zero indices](https://github.com/librosa/librosa/blob/d9b72af964820fc796425c55d3dccd7f6b908bf9/librosa/core/spectrum.py#L417):
```
approx_nonzero_indices = ifft_window_sum > util.tiny(ifft_window_sum)
y[approx_nonzero_indices] /= ifft_window_sum[approx_nonzero_indices]
```

@mthrok Shall we make changes to ``istft`` in the same way as librosa?